PV260_Project/PresentationLayer/App.cs

```diff
@@ -22,6 +28,9 @@ public class App : IApp
                         break;
                     case "2":
                         _emailUi.Run();
+                        break;
+                    case "3":
+                        _configUi.Run();
```

Comment:
The `IUi.Run()` method is asynchronous, therefore `_emailUi.Run()` (and other method calls) should be awaited I assume? Then `IApp.Run()` needs to be return `Task` as well, and some slight changes also have to be made in `Program.cs`, where `App` is used.

Reply:
Good point, I guess there is no big deal of it being executed synchronously, but It's better using the `await` when we implemented it as async.

Reply:
this is more of a general change so I would not put it in this branch since it can also cause conflicts, I can make the modifications after merge